docs/_data_templates/configuration-parameters.j2

```diff
@@ -1,29 +1,29 @@
-The configuration parameters will be set by the :doc:`Nornir.core.configuration.Config </ref/api/configuration>` class.
+{% macro document_section(section, schema) %}
+{{ section }}
+{{ "-" * section|length }}
 
-{% for k, v in params|dictsort %}
-----------
+{% for k, v in schema["properties"].items() -%}
 
-{{ k }}
-----------------------------------
+``{{ k }}``
+{{ "_" * (k|length + 4) }}
 
+.. list-table::
+   :widths: 15 85
 
-.. raw:: html
+   * - **Descrpiption**
+     - {{ v["description"] }}
+   * - **Type**
+     - ``{{ v["type"] }}``
+   * - **Default**
+     - {{ "``{}``".format(v["default"]) if v["default"] else "" }}
+   * - **Required**
+     - ``{{ v["required"] }}``
+   * - **Environment Variable**
+     - ``{{ "NORNIR_{}_{}".format(section, k).upper() }}``
```

Comment:
I haven't looked at the entire PR, along with all of the changes to Nornir that have been made while I haven't been paying attention. However, my idea for this was to write out "N/A" for the variable types which take a list or dictionary as those can't be set through environment variables. Just seeing at this it looks like all settings can be configured using environment variables. So, that's why the if statement which is removed is included.

Reply:
As per pydantic documentation:

> Complex types like list, set, dict and submodels can be set by using JSON environment variables.

So it should work.

Reply:
Ah, wasn't aware that was possible.